File: src/main/scala/cub/hackerrank/challenges/algorithms/easy/timeConversion/timeConversion.js
const fs = require('fs')

process.stdin.resume()
process.stdin.setEncoding('utf-8')

let inputString = ''
let currentLine = 0

process.stdin.on('data', (inputStdin) => {
  inputString += inputStdin
})

process.stdin.on('end', (_) => {
  inputString = inputString
    .trim()
    .split('\n')
    .map((str) => str.trim())

  main()
})

function readLine() {
  // eslint-disable-next-line
  return inputString[currentLine++]
}

/*
 * Complete the timeConversion function below.
 */
function timeConversion(s) {
  const time12 = s.toUpperCase()
  const [_, hh, rest, ampm] = time12.match(/^(..)(.*)(..)$/)
  let h = parseInt(hh, 10)
  if (ampm === 'PM') {
    if (h === 12) {
      h = 12
    } else {
      h += 12
    }
  } else if (h === 12) {
    h = 0
  }
  return h.toString().padStart(2, '0') + rest
}

function main() {
  const ws = fs.createWriteStream(process.env.OUTPUT_PATH)
  const s = readLine()
  const result = timeConversion(s)
  ws.write(`${result}\n`)
  ws.end()
}

File: src/main/java/cub/hackerrank/challenges/algorithms/easy/timeConversion/timeConversion.py
#!/bin/python3

import os
import sys


#
# Complete the timeConversion function below.
#
def timeConversion(s):
    time12 = s.upper()
    (h, rest, ampm) = (int(time12[0:2]), time12[2:-2], time12[-2:])
    if ampm == "AM":
        return "%02d%s" % (0 if h == 12 else h, rest)
    else:
        return "%02d%s" % (12 if h == 12 else h + 12, rest)


#
# Write your code here.
#

if __name__ == '__main__':
    f = open(os.environ['OUTPUT_PATH'], 'w')
    s = input()
    result = timeConversion(s)
    f.write(result + '\n')
    f.close()
